tests/bluetooth/bsim_bt/bsim_test_mesh/src/settings_test_backend.c

```diff
@@ -21,6 +21,7 @@
 LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 
 #define SETTINGS_FILE "settings_data.log"
+#define SETTINGS_FILE_TMP "~settings_data.log"
```

Comment:
Why do you need the temporary file? What was wrong with the single file approach?

Reply:
It was corrupting the data. AFAIK there is no way to replace text in file using single file approach.

Reply:
Perhaps the easiest approach would just be to load everything into a buffer, edit it, then rewrite the entire file? This appears to work as well though, but it'd be helpful to add some explanatory comment for this mechanism, as it looks like an odd design decision on the surface

Reply:
The file can be too big to keep it in memory. Also, since `fflush` is not called directly, the buffer will _most likely_ be flushed upon `ffclose` call.

Reply:
I liked the temporary file idea, it is flexible and it keeps the backend free of buffers that need to be future proofed. Having a local buffer would create maintenance problems if someone happens to stress test the device with a large amount of persistence data, the test will fail, and buffer sizes would have to be increased.

Reply:
It is run on PC (or another POSIX compatible platform). That has more than enough memory to keep there settings file. Kind of agree with @trond-snekvik approach with buffer looks less magic.